### PR TITLE
Add filename components in Checkpoint

### LINKED ISSUE #2480: Pass filename components into Checkpoint.load_objects
## 🚀 Feature

Currently, objects can be loaded like this:

```python
from ignite.handlers import Checkpoint

Checkpoint.load_objects(to_load=to_load, checkpoint=checkpoint_fp)
```

It would be nice to have the following as an option:
```python
from ignite.handlers import Checkpoint

checkpointer = Checkpoint(...)
checkpointer .load_objects(to_load=to_load, filename_components={"name": name, "global_step":global_step})
```

In other words, pass in components of ```self.filename_pattern```. The Checkpoint object already has ```self.filename_pattern```, ```self.filename_prefix```, ```self.ext```, and it should also have the ```dirname``` that is passed into ```__init__```. So most of the components of constructing the filename are already defined in the Checkpoint object. Getting the user to recreate the full file path is unnecessary.

### PR DESCRIPTION
**Pull request by sdesrozis**
Fixes #2480 

Description:

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -327,6 +327,18 @@ def __init__(
         self.include_self = include_self
         self.greater_or_equal = greater_or_equal
 
+    def __get_filename_pattern(self, global_step: Optional[int]) -> str:
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:330]: @sdesrozis why using dunder `__` instead of single underscore `_` ,  `_get_filename_pattern` ?

**Review by vfdev-5 — COMMENTED** [review]
Looks good to me, just a nit about refactored function name

File: ignite/handlers/checkpoint.py (hunk 1 above)
Comment (sdesrozis) [reply ignite/handlers/checkpoint.py:330]: For privacy but one underscore would be enough.

File: ignite/handlers/checkpoint.py [hunk 2]
```diff
@@ -519,41 +523,51 @@ def _check_objects(objs: Mapping, attr: str) -> None:
...
+
+                    to_save = {"weights": model, "optimizer": optimizer}
+
+                    trainer.add_event_handler(Events.EPOCH_COMPLETED(every=2), handler, to_save)
+                    trainer.run(torch.randn(10, 1), 5)
+
+                    to_load = to_save
+                    checkpoint_fp = Path(tmpdirname) / 'myprefix_checkpoint_40.pt'
+                    checkpoint = torch.load(checkpoint_fp)
+                    Checkpoint.load_objects(to_load=to_load, checkpoint=checkpoint)
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:564]: @sdesrozis this example is not working: 
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [2], in <cell line: 10>()
     21 to_load = to_save
     22 # load checkpoint myprefix_checkpoint_40.pt
---> 23 checkpoint.load_objects(to_load=to_load, global_step=40)

TypeError: load_objects() missing 1 required positional argument: 'checkpoint'
```

File: ignite/handlers/checkpoint.py [hunk 3]
```diff
@@ -519,41 +523,51 @@ def _check_objects(objs: Mapping, attr: str) -> None:
...
         Examples:
             .. code-block:: python
 
+                import tempfile
+                from pathlib import Path
+
                 import torch
+
                 from ignite.engine import Engine, Events
                 from ignite.handlers import ModelCheckpoint, Checkpoint
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:546]: We can also remove unused Checkpoint

**Pull request merged**